rdt-app/src/main/java/io/ona/rdt_app/widget/RDTBarcodeFactory.java

```diff
@@ -78,7 +80,11 @@ public void onActivityResult(int requestCode, int resultCode, Intent data) {
                                     try {
                                         Barcode barcode = data.getParcelableExtra(JsonFormConstants.BARCODE_CONSTANTS.BARCODE_KEY);
                                         Log.d("Scanned QR Code", barcode.displayValue);
-                                        jsonObject.put(VALUE, barcode.displayValue.split(",")[0]);
+                                        String[] barcodeValues = barcode.displayValue.split(",");
+                                        String idAndExpDate = barcodeValues[0] + "," + barcodeValues[1];
+                                        jsonObject.put(VALUE, idAndExpDate);
+                                        jsonApi.writeValue("step5", Constants.Form.LBL_RDT_ID, barcodeValues[0], "", "", "", false);
+                                        jsonApi.writeValue("step2", Constants.Form.EXPIRATION_DATE_READER, barcodeValues[1], "", "", "", false);
```

Comment:
Same as above, this this a good idea hardcoding steps and editing other fields from a factory in terms on forms editing